### PR TITLE
feat: normalize search query

### LINKED ISSUE #154: normalize tags and author keys
### Problem

Currently tags and authors are case sensitive and expressed in plain text resulting in ugly/fragile preview urls:

`?author=Kyle%20Drake&tags=go-ipfs,bootstrap%20nodes,breaking%20change`

### Suggestion:

Use this included `slug` module to enforce urls into lower kebab-case keys and match the objects in a case insensitive way:

`?author=kyle-drake&tags=go-ipfs,bootstrap-nodes,breaking-change`

It would be best to do this sooner rather than later so that we don't end up with too many inbound links using the current structure.

### PR DESCRIPTION
**Pull request by jdiogopeixoto**
- Normalizes all the fields in the search query, except the `search`
- Keeps support for the old format of categories - accessing `/?category=Blog%20post` redirects to `?/category=blog-post`

Closes #154

### CODE REVIEW

File: src/.vuepress/theme/layouts/Blog.vue [hunk 1]
```diff
@@ -183,25 +194,20 @@ export default {
...
-          uniq(
-            data
-              .filter((subPage) => subPage.tags)
-              .map((subPage) => subPage.tags)
-              .flat(2)
-          )
-        )
+        data.forEach((subPage) => {
+          if (subPage.tags) {
+            subPage.tags.map((tag) => tagsArray.push(tag))
```
Comment (zebateira) [inline src/.vuepress/theme/layouts/Blog.vue:199]:
```suggestion
            tagsArray.push(...subPage.tags)
```

**Review by zebateira — CHANGES_REQUESTED** [review]
Nice job here! 👌

**Review by zebateira — APPROVED** [review]
👌

**Review by jessicaschilling — APPROVED** [review]
If @zebateira and @cwaring are good, I am too 😊  Thanks!

**Comment by zebateira** [comment]
just added the change to redirect the category routes, so now `/events` should redirect to `/?category=event`.
all redirects:

- /events → /?category=event
- /videos → /?category=video
- /tutorials → /?category=tutorial
- /releasenotes → /?category=release-notes
- /newscoverage → /?category=news-coverage
- /academicpapers → /?category=academic-paper

**Comment by jessicaschilling** [comment]
@zebateira @jdiogopeixoto -- how confident are you in whether it would be permissible to merge this as-is and address any future comments from @cwaring in a separate issue?

**Comment by zebateira** [comment]
@jessicaschilling yeah sounds good 👍

**Comment by cwaring** [comment]
Thanks @jdiogopeixoto, great implementation 👏🏻

**Pull request merged**